qt/component-box.cpp

```diff
@@ -129,3 +136,22 @@ void ComponentBox::sortByScore()
 {
     as_component_box_sort_by_score(d->m_cbox);
 }
+
+void ComponentBox::operator+=(const ComponentBox& other)
+{
+    for (uint i = 0; i < as_component_box_len(other.d->m_cbox); i++) {
+        g_autoptr(GError) error = nullptr;
+        AsComponent *c = as_component_box_index(other.d->m_cbox, i);
+        as_component_box_add(d->m_cbox, c, &error);
+        if (error) {
+            qWarning() << "error adding component" << error->message;
+        }
+    }
+}
+
+ComponentBox::iterator ComponentBox::erase(iterator it)
+{
+    GPtrArray *array = as_component_box_array(it.data->d->m_cbox);
+    g_ptr_array_remove_index(array, it.index);
```

Comment:
You *must not* do this, it will break the component box (the returned box is constant, but unfortunately C/C++ will let you do this even though it shouldn't be done). Use `as_component_box_remove_at` instead.